packages/core/src/Nodes/FunctionNode.ts

```diff
@@ -26,20 +27,72 @@ export class FunctionNode extends Node {
     );
   }
 }
-export class FunctionNode2 extends FunctionNode {
+
+export class FunctionDesc extends NodeDescription {
   constructor(props: {
-    description: NodeDescription;
-    graph: Graph;
-    inputs?: Socket[];
-    outputs?: Socket[];
-    exec: () => void;
+    name: string;
+    label?: string;
+    category?: NodeCategory;
+    aliases?: string[];
+    in?: { [name: string]: string }[] | string[];
+    out: { [name: string]: string }[] | string;
+    exec: (...args: any[]) => any;
   }) {
```

Comment:
This whole code block is kinda long and hard to read - could you please break into smaller parts and/or add some comments explaining whats happening?